passthrough-server/src/main/java/org/terracotta/passthrough/PassthroughServerProcess.java

```diff
@@ -709,6 +709,12 @@ public synchronized void create(String entityClassName, String entityName, long
       // Tell the entity to create itself as something new.
       newEntity.createNew();
```

Comment:
This call happens after the entity has been added to the entity map, so cleanup required in the exception path.